app/serializers/user_bookmark_serializer.rb

```diff
@@ -109,7 +109,13 @@ def raw
   end
 
   def cooked
-    post.cooked
+    @cooked ||= \
+      if object.for_topic && last_read_post_number.present?
+        post_number = [last_read_post_number + 1, highest_post_number].min
```

Comment:
I was curious about `last_read_post_number + 1` and made an experiment about splitting topic. 

In that case, you may hit post about split:

<img width="786" alt="Screen Shot 2021-09-22 at 10 23 37 am" src="https://user-images.githubusercontent.com/72780/134264645-5605ac2b-7a98-4d86-be2b-94c66b1e29d4.png">

It might be ok as making calculation to next "normal" post will be more expensive. What do you think?

Reply:
Oh this is a good point...I just remembered there is an `Unread` class to do all this calculation, I will use that

Reply:
we can use date and find next post > read_at && "normal"

Reply:
Wait that is just a count...I will try and find a nicer way anyway

Reply:
We can't use a date, we always use `last_read_post_number` for these. So I will say `Post.where("post_number > last_read_post_number").where(type: regular)`